### PR TITLE
Emit failover events into the job autoscaler events to allow autoscaling decisions

### PR DESCRIPTION
**Pull request by hmitnflx**
### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/JobAutoScaler.java [hunk 1]
```diff
@@ -341,7 +341,12 @@ public int getDesiredWorkersForScaleUp(final int increment, final int numCurrent
           return numCurrentWorkers;
         } else {
           final int maxWorkersForStage = stageSchedulingInfo.getScalingPolicy().getMax();
-          desiredWorkers = Math.min(numCurrentWorkers + increment, maxWorkersForStage);
+          if (reason == ScalingReason.AutoscalerManager) {
```
Comment (hmitnflx) [inline mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/JobAutoScaler.java:344]: todo: this condition is no longer correct.

File: mantis-common/src/main/java/io/mantisrx/runtime/descriptor/StageScalingPolicy.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ public enum ScalingReason {
         RPS,
         JVMMemory,
         SourceJobDrop,
-        FailoverAware
+        AutoscalerManager
```
Comment (Andyz26) [inline mantis-common/src/main/java/io/mantisrx/runtime/descriptor/StageScalingPolicy.java:89]: AutoscalerManager vs RPS/JVMMemory seems a bit strange?

File: mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/JobAutoscalerManager.java [hunk 3]
```diff
@@ -39,6 +39,10 @@ default boolean isScaleDownEnabled() {
         return true;
     }
 
+    default double getCurrentValue() {
```
Comment (Andyz26) [inline mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/JobAutoscalerManager.java:42]: can you add a document here about why it's -1?

File: mantis-common/src/main/java/io/mantisrx/runtime/descriptor/StageScalingPolicy.java (hunk 2 above)
Comment (Andyz26) [reply mantis-common/src/main/java/io/mantisrx/runtime/descriptor/StageScalingPolicy.java:89]: maybe "AutoscalerManagerEvent"

File: mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/WorkerMetricHandler.java [hunk 4]
```diff
@@ -416,6 +417,19 @@ public void onNext(MetricData metricData) {
             };
         }
 
+        private void maybeEmitAutoscalerManagerEvent(int numWorkers) {
+            final double currentValue = jobAutoscalerManager.getCurrentValue();
+            if (currentValue >= 0.0 && currentValue <= 100.0) {
```
Comment (Andyz26) [inline mantis-runtime-executor/src/main/java/io/mantisrx/server/worker/jobmaster/WorkerMetricHandler.java:422]: can you add a inline doc about this logic? why it's 0 to 100?

**Review by Andyz26 — APPROVED** [review]

**Pull request merged**